### PR TITLE
Idva6 761 fetch associations for company with email

### PR DESCRIPTION
**Pull request by krishna-patel-ch**
__Short description outlining key changes/additions__

Changes were made to the /associations/companies/{company_number} endpoint. The endpoint now has an additional, optional filter that can be applied based on the user's email address.

__JIRA Ticket Number__

IDVA6-761

### Type of change

* [ ] Bug fix
* [x] New feature
* [ ] Breaking change
* [ ] Infrastructure change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [x] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/accounts/association/repositories/AssociationsRepository.java [hunk 1]
```diff
@@ -28,4 +28,8 @@ public interface AssociationsRepository extends MongoRepository<AssociationDao,
     @Query( value = "{ 'company_number': ?0, 'user_id': ?1 }", exists = true )
     boolean associationExists( String companyNumber, String userId );
 
+
+    @Query( "{ 'company_number': ?0, 'user_email': ?1, 'status': { $in: ?2 } } }" )
+    Page<AssociationDao> fetchAssociation( final String companyNumber, final String userEmail, final Set<String> statuses, final Pageable pageable );
```
Comment (kkonuganti-ch) [inline src/main/java/uk/gov/companieshouse/accounts/association/repositories/AssociationsRepository.java:33]: change the name to be more specific based on query.

**Review by kkonuganti-ch — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/companieshouse/accounts/association/repositories/AssociationsRepository.java (hunk 1 above)
Comment (krishna-patel-ch) [reply src/main/java/uk/gov/companieshouse/accounts/association/repositories/AssociationsRepository.java]: Thanks, I have now renamed the method to be more descriptive

**Review by kkonuganti-ch — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Pull request merged**